pandas/core/tools/datetimes.py

```diff
@@ -179,14 +180,14 @@ def _convert_listlike_datetimes(arg, box, format, name=None, tz=None,
 
     # these are shortcutable
     if is_datetime64tz_dtype(arg):
-        if not isinstance(arg, DatetimeIndex):
+        if not isinstance(arg, (DatetimeArray, DatetimeIndex)):
             return DatetimeIndex(arg, tz=tz, name=name)
         if tz == 'utc':
             arg = arg.tz_convert(None).tz_localize(tz)
         return arg
 
     elif is_datetime64_ns_dtype(arg):
-        if box and not isinstance(arg, DatetimeIndex):
+        if box and not isinstance(arg, (DatetimeArray, DatetimeIndex)):
```

Comment:
are there tests specific to this and the change on L183?

Reply:
This is necessary because DatetimeArray is no longer a subclass of DatetimeIndex. I don't think you could construct a test to hit it that isn't redundant.

Reply:
I wasn't able to construct an example that hits this.